src/App.tsx

```diff
@@ -17,18 +66,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                filterSelect={filterSelect}
+                setFilterSelect={setFilterSelect}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+               {isLoading && todos.length === 0 ? <Loader />
+                : (
+                <TodoList
+                  todos={getFilteredTodos()}
```

Comment:
Running this filtering on every render can slow down your app. Improve this by memoizing this value (by using `useMemo` in component)